lib/stb_image.ex

```diff
@@ -93,23 +109,61 @@ defmodule StbImage do
   @doc """
   Decode image from buffer in memory
 
-  - **buffer**. Buffered raw file data in memory.
-  - **desired_channels**. `0` for auto-detection. Otherwise, the number of desired channels.
-  - **type**. Specify format for each channel. `:u8`, `:u16` or `:f32`.
+  - `buffer`. Buffered raw file data in memory.
+  - `desired_channels`. `0` for auto-detection. Otherwise, the number of desired channels.
+  - `type`. Specify format for each channel. `:u8`, `:u16` or `:f32`.
 
   ## Example
-  ```elixir
-  # image buffer from a file or perhaps download from Internet
-  {:ok, buffer} = File.read("/path/to/image")
-  # decode the image from memory
-  # and specify it is a 3-channel image and each channel is in uint8_t
-  {:ok, img, shape, type} = StbImage.from_memory(buffer, 3, :u8)
-  {h, w, c} = shape
-  ```
+
+      # image buffer from a file or perhaps download from Internet
+      {:ok, buffer} = File.read("/path/to/image")
+      # decode the image from memory
+      # and specify it is a 3-channel image and each channel is in uint8_t
+      {:ok, img, shape, type, channels} = StbImage.from_memory(buffer, 3, :u8)
+      {h, w, c} = shape
+
   """
   def from_memory(buffer, desired_channels, type)
-      when is_binary(buffer) and desired_channels >= 0 and
-             (type == :u8 or type == :u16 or type == :f32) do
+      when is_binary(buffer) and is_integer(desired_channels) and desired_channels >= 0 and
+             type in @types do
     StbImage.Nif.from_memory(buffer, desired_channels, type)
   end
+
+  @doc """
+  Decode GIF image from a given file
+
+  - `filename`. Path to the GIF image.
+
+  ## Example
+
+      {:ok, frames, shape, delays} = StbImage.gif_from_file("/path/to/image")
+      {h, w, 3} = shape
+      # GIFs always have channels == :rgb and type == :u8
+      # delays is a list that has n elements, where n is the number of frames
+
+  """
+  def gif_from_file(filename) do
+    with {:ok, buffer} <- File.read(filename) do
+      gif_from_memory(buffer)
+    end
+  end
+
+  @doc """
+  Decode image from buffer in memory
+
+  - `buffer`. Path to the image.
+
+  ## Example
+
+      {:ok, buffer} = File.read("/path/to/image")
+      {:ok, frames, shape, delays} = StbImage.gif_from_memory(buffer)
+      {h, w, 3} = shape
+      # delays is a list that has n elements, where n is the number of frames
+
+  """
+  def gif_from_memory(buffer), do: StbImage.Nif.gif_from_memory(buffer)
+
+  def to_file(filename, extension, data, width, height, channels) do
```

Comment:
Please add some docmentation and some guards. Something like:

```suggestion
  def to_file(filename, extension, data, width, height, channels) when is_binary(filename) and extension in ~w(jpg png ...) and ... do
    ...
```